src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -33,162 +43,465 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+            //
+            // Install dependencies:
+            // > sudo apt install hunspell-tools hunspell-en-us
+            // 
+            // Create dictionary:
+            // > unmunch /usr/share/hunspell/en_US.dic /usr/share/hunspell/en_US.aff > Dictionary.dic
+            //
+            // Tweak:
+            // Added the words: 'namespace'
+            var text = SourceText.From(TextAnalyzersResources.Dictionary);
+            return ParseDicDictionary(text);
+        }
+
+        private static CodeAnalysisDictionary ParseXmlDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromXml);
+
+        private static CodeAnalysisDictionary ParseDicDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromDic);
+
+        private static string RemovePrefixIfPresent(string prefix, string name)
+            => name.StartsWith(prefix, StringComparison.Ordinal) ? name[1..] : name;
+
+        private static Diagnostic GetMisspelledWordDiagnostic(ISymbol symbol, string misspelledWord)
+        {
+            return symbol.Kind switch
+            {
+                SymbolKind.Assembly => Diagnostic.Create(AssemblyRule, Location.None, misspelledWord, symbol.Name),
+                SymbolKind.Namespace => Diagnostic.Create(NamespaceRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString()),
+                SymbolKind.NamedType => Diagnostic.Create(TypeRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString()),
+                SymbolKind.Method or SymbolKind.Property or SymbolKind.Event or SymbolKind.Field => Diagnostic.Create(MemberRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString()),
+                SymbolKind.Parameter => symbol.ContainingType.TypeKind == TypeKind.Delegate
+                    ? Diagnostic.Create(DelegateParameterRule, symbol.Locations.First(), symbol.ContainingType.ToDisplayString(), misspelledWord, symbol.Name)
+                    : Diagnostic.Create(MemberParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name),
+                SymbolKind.TypeParameter => symbol.ContainingSymbol.Kind == SymbolKind.Method
+                    ? Diagnostic.Create(MethodTypeParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name)
+                    : Diagnostic.Create(TypeTypeParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name),
+                SymbolKind.Local => Diagnostic.Create(VariableRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString()),
+                _ => throw new NotImplementedException($"Unknown SymbolKind: {symbol.Kind}"),
+            };
+        }
+
+        private static Diagnostic GetUnmeaningfulIdentifierDiagnostic(ISymbol symbol, string symbolName)
```

Comment:
You could use expression body here instead. Also please apply my previous suggestions here too.

Reply:
📝 Edited this request to reduce work per https://github.com/dotnet/roslyn-analyzers/pull/1111#discussion_r548020530

Reply:
I assume this is outdated